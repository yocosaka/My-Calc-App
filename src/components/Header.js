import React from 'react';
import { NavLink } from 'react-router-dom';

const Header = () => (
  <header className="header">
    <h1 className="header__title">Math Magicians</h1>
    <nav className="header__nav">
      <NavLink exact to="/">Home</NavLink>
      <NavLink exact to="/calculator">Calculator</NavLink>
      <NavLink exact to="/quotes">Quotes</NavLink>
    </nav>
  </header>
);

export default Header;
