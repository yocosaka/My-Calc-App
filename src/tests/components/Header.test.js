import React from 'react';
// import { render, screen } from '@testing-library/react';

import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import Header from '../../components/Header';

// describe('Header', () => {
//   test('renders Header component ', () => {
//     render(<Header />);
//     // screen.debug();
//     // eslint-disable-next-line no-unused-expressions
//     expect(screen.getByText('Yoko Saka')).toBeInTheDocument;
//   });
// });

describe('Test suits for Header with Link', () => {
  test('should match with snapshot', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Header />
        </BrowserRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
