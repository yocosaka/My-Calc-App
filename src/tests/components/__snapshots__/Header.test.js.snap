// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test suits for Header with Link should match with snapshot 1`] = `
<header
  className="header"
>
  <h1
    className="header__title"
  >
    Math Magicians
  </h1>
  <nav
    className="header__nav"
  >
    <a
      aria-current="page"
      className="active"
      href="/"
      onClick={[Function]}
      style={Object {}}
    >
      Home
    </a>
    <a
      aria-current={null}
      href="/calculator"
      onClick={[Function]}
    >
      Calculator
    </a>
    <a
      aria-current={null}
      href="/quotes"
      onClick={[Function]}
    >
      Quotes
    </a>
  </nav>
</header>
`;
